### PR TITLE
Add bootstrap VPC `SecurityGroup` resource support

### PR DESCRIPTION
**Pull request by michaelhtm**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/acktest/bootstrapping/vpc.py [hunk 1]
```diff
@@ -144,6 +144,50 @@ def cleanup(self):
...
+        return boto3.resource("ec2", region_name=self.region)
+
+    def bootstrap(self):
+        """Creates security group with an auto-generated name and description.
+        """
+        vpc = self.ec2_resource.VPC(self.vpc_id)
+        group = vpc.create_security_group(
+            Description=self.description,
+            GroupName=self.name,
+        )
```
Comment (a-hilaly) [inline src/acktest/bootstrapping/vpc.py:178]: I think we need to set the vpc ID here, other wise it's gonna be created against the default VPC. https://aws-controllers-k8s.github.io/community/reference/ec2/v1alpha1/securitygroup/#spec

File: src/acktest/bootstrapping/vpc.py [hunk 2]
```diff
@@ -144,6 +144,50 @@ def cleanup(self):
...
+        self.arn = "arn:aws:ec2:{region}:{accId}:security-group/{sgId}".format(region=self.region, accId=self.account_id, sgId=self.group_id)
+
+    def cleanup(self):
+        """Deletes the subnets.
+        """
+        # You must delete the securityGroup before you can delete any of its dependencies
+        self.ec2_client.delete_security_group(
+            GroudId=self.group_id,
+            GroupName=self.name,
+        )
```
Comment (a-hilaly) [inline src/acktest/bootstrapping/vpc.py:189]: ditto

File: src/acktest/bootstrapping/vpc.py (hunk 1 above)
Comment (michaelhtm) [reply src/acktest/bootstrapping/vpc.py:178]: vpc ID is not required if i'm creating it using this: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/vpc/create_security_group.html

Comment (a-hilaly) [reply src/acktest/bootstrapping/vpc.py:178]: ohhh you're right!

**Review by a-hilaly — APPROVED** [review]
/lgtm

**Pull request merged**